openfeature/client.go

```diff
@@ -38,22 +38,29 @@ type IClient interface {
 
 // ClientMetadata provides a client's metadata
 type ClientMetadata struct {
-	name string
+	domain string
+	name   string
```

Comment:
we don't have to change the internal values yet, the getter `Domain()` mapping to the old internal field is enough. 

  ```suggestion
	name   string
```